### DISCUSSION TITLE
Back button after deleting a branch

### DISCUSSION THREAD

**Discussion by Its-Just-Nans** [community/community#28104] [Other Feature Feedback, Questions, & Ideas]
# Suggestion to counter a "bug"

## Bug

How to reproduce 
- have a github repo
- checkout to a branch - e.g. `https://github.com/community/community/tree/jay-patch-1`
- go on the `branchs` tabs - e.g. `https://github.com/community/community/branches`
- delete the branch - e.g. `jay-patch-1`
- go to the previous page
- you get a 404

## Suggestion

- detect if the branch is still present - e.g. `jay-patch-1`
- redirect to the repo page - e.g. `https://github.com/community/community`

**Comment by airtower-luna** [comment]
> redirect to the repo page - e.g. https://github.com/community/community

I think that'd run a high risk of confusing users: There isn't a lot of visual difference between the repository page and opening any branch – mostly just the selected branch drop-down list and the "N commits ahead of/behind the default branch" box. Someone might think they actually got the branch they wanted, until they run into trouble later. :sweat_smile: 

404 is the right response for trying to access something that no longer exists. A hint like "do you want to visit [repository]?" might be nice, but also adds complexity.

**Reply by Its-Just-Nans** [reply]
The "bug" appears when you delete a branch.

And you delete a branch, you know that the branch is gone.

**Comment by Casm101** [comment]
It's not exactly a bug in this case.

Your idea is good in the case you where dealing with an application or a piece of software that does't use static routes, but in the case of GitHub, a website with static links, a 404 for: page doesn't exist, or a 410 for: resource no longer available is a bit more appropriate.

Yes the branch has been deleted, but so has the static link for it, and without messing around some weird rerouting, its a bit of a pointless task.
